src/main/java/org/gradle/playframework/plugins/PlayApplicationPlugin.java

```diff
@@ -131,7 +134,42 @@ private TaskProvider<Jar> createAssetsJarTask(Project project) {
 
     private void registerOutgoingArtifact(Project project, TaskProvider<Jar> assetsJarTask) {
         Configuration runtimeElementsConfiguration = project.getConfigurations().getByName(RUNTIME_ELEMENTS_CONFIGURATION_NAME);
-        PublishArtifact jarArtifact = new LazyPublishArtifact(assetsJarTask);
+        PublishArtifact jarArtifact = new PublishArtifact() {
+            @Override
+            public String getName() {
+                return assetsJarTask.get().getArchiveFileName().get();
+            }
+
+            @Override
+            public String getExtension() {
+                return "jar";
+            }
+
+            @Override
+            public String getType() {
+                return "jar";
+            }
+
+            @Override
+            public String getClassifier() {
+                return assetsJarTask.get().getArchiveClassifier().get();
+            }
+
+            @Override
+            public File getFile() {
+                return assetsJarTask.get().getArchiveFile().get().getAsFile();
+            }
+
+            @Override
+            public Date getDate() {
+                return new Date();
+            }
+
+            @Override
+            public TaskDependency getBuildDependencies() {
+                return task -> Collections.singleton(assetsJarTask.get());
+            }
+        };
         ConfigurationPublications publications = runtimeElementsConfiguration.getOutgoing();
         publications.getArtifacts().add(jarArtifact);
```

Comment:
Can you try `publications.artifact(assetsJarTask);` which removes the need for an implementation of `PublishArtifact` entirely?